runtime/compiler/optimizer/JProfilingValue.cpp

```diff
@@ -795,10 +859,17 @@ TR::Node *
 TR_JProfilingValue::createHelperCall(TR::Compilation *comp, TR::Node *value, TR::Node *table)
    {
    TR::SymbolReference *profiler;
-   if (value->getDataType() == TR::Int32)
+   TR::DataType valueType = value->getDataType();
+   if (value->getSize() <=4)
+      {
+      if (value->getDataType() != TR::Address)
```

Comment:
I think this at least warrants a comment

Reply:
Added comment in https://github.com/eclipse/openj9/pull/6585/commits/4babc9149b99cc98d9703d5340c8c21f81c918da